microsoft-edge/webview2/reference/javascript/webview.yml

```
@@ -0,0 +1,386 @@
+              HRESULT CallCallbackAsynchronously([in] IDispatch* callbackParameter);
+
+              // Demonstrate a property which uses Date types
+              [propget] HRESULT DateProperty([out, retval] DATE * dateResult);
+              [propput] HRESULT DateProperty([in] DATE dateValue);
+
+              // Creates a date object on the native side and sets the DateProperty to it.
+              HRESULT CreateNativeDate();
+          };
+
+          ```
+
+
+          Add an instance of this interface into your JavaScript with `AddHostObjectToScript`. In this case, name it
+          `sample`.
+
+
+          In native host app code:
+
+
+          ```cpp
+
+          VARIANT remoteObjectAsVariant = {};
+
+          m_hostObject.query_to<IDispatch>(&remoteObjectAsVariant.pdispVal);
+
+          remoteObjectAsVariant.vt = VT_DISPATCH;
+
+
+          // We can call AddHostObjectToScript multiple times in a row without
+
+          // calling RemoveHostObject first. This will replace the previous object
+
+          // with the new object. In our case this is the same object and everything
+
+          // is fine.
+
+          CHECK_FAILURE(
+              m_webView->AddHostObjectToScript(L"sample", &remoteObjectAsVariant));
+          remoteObjectAsVariant.pdispVal->Release();
+
+          ```
+
+
+          In the HTML document, use the COM object using `chrome.webview.hostObjects.sample`.
+
+
+          ```html
+
+          document.getElementById("getPropertyAsyncButton").addEventListener("click", async () => {
+              const propertyValue = await chrome.webview.hostObjects.sample.property;
+              document.getElementById("getPropertyAsyncOutput").textContent = propertyValue;
+          });
+
+
+          document.getElementById("getPropertySyncButton").addEventListener("click", () => {
+              const propertyValue = chrome.webview.hostObjects.sync.sample.property;
+              document.getElementById("getPropertySyncOutput").textContent = propertyValue;
+          });
+
+
+          document.getElementById("setPropertyAsyncButton").addEventListener("click", async () => {
+              const propertyValue = document.getElementById("setPropertyAsyncInput").value;
+              // The following line will work but it will return immediately before the property value has actually been set.
+              // If you need to set the property and wait for the property to change value, use the setHostProperty function.
+              chrome.webview.hostObjects.sample.property = propertyValue;
+              document.getElementById("setPropertyAsyncOutput").textContent = "Set";
+          });
+
+
+          document.getElementById("setPropertyExplicitAsyncButton").addEventListener("click", async () => {
+              const propertyValue = document.getElementById("setPropertyExplicitAsyncInput").value;
+              // If you care about waiting until the property has actually changed value use the setHostProperty function.

```

```suggestion
              // If you care about waiting until the property has actually changed value, use the setHostProperty function.
```
comma like previous quote has